thrifty-schema/src/main/java/com/microsoft/thrifty/schema/FieldNamingPolicy.java

```diff
@@ -58,4 +77,42 @@ public String apply(String name) {
             return name;
         }
     };
+
+    /**
+     * Find case format from string.
+     * @param s the input String
+     * @return CaseFormat the case format of the string.
+     */
+    @Nullable
+    private static CaseFormat CASE_FORMAT(String s) {
+
+        if (s.contains("_")) {
+
+            if (s.toUpperCase().equals(s))
+                return CaseFormat.UPPER_UNDERSCORE;
+
+            if (s.toLowerCase().equals(s))
+                return CaseFormat.LOWER_UNDERSCORE;
+
+        } else if (s.contains("-")) {
+
+            if (s.toLowerCase().equals(s))
+                return CaseFormat.LOWER_HYPHEN;
+
+        } else {
+
+            if (Character.isLowerCase(s.charAt(0))) {
+
+                if (s.matches("([a-z]+[A-Z]+\\w+)+"))
```

Comment:
This will be invoked potentially thousands of times - might as well pull the expression out into a `Pattern.compile`'d static member.

Reply:
Good point. https://github.com/Microsoft/thrifty/pull/165/commits/2472d844ade0a6b2d5a9def89e26dfef1f322e62